smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/PackageJsonGenerator.java

```diff
@@ -43,9 +43,19 @@ static void writePackageJson(
     ) {
         // Write the package.json file.
         InputStream resource = PackageJsonGenerator.class.getResourceAsStream("base-package.json");
-        ObjectNode node = Node.parse(IoUtils.toUtf8String(resource))
-                        .expectObjectNode()
-                        .merge(settings.getPackageJson());
+
+        ObjectNode userSuppliedPackageJson = settings.getPackageJson();
+        ObjectNode defaultPackageJson = Node.parse(IoUtils.toUtf8String(resource))
+            .expectObjectNode();
+
+        ObjectNode mergedScripts = defaultPackageJson.expectObjectNode("scripts")
+            .merge(
```

Comment:
Is the merge on line 57 not sufficient for this?

Reply:
It's a shallow merge, so this does the following:

Default:
```json
{
  "name": "A",
  "scripts": {
    "build": "..."  
  },
  "metadata": "A"
}
```

User supplied:
```json
{
  "name": "B",
  "scripts": {
    "test": "..."  
  }
}
```

- create the merged `pkg.scripts` object.
```json
{
  "build": "...",
  "test": "..."
}
```
- do the normal shallow merge
```json
{
  "name": "B",
  "scripts": {
    "test": "..."  
  },
  "metadata": "A"
}
```
- write the merged `pkg.scripts` onto the shallow merge
```json
{
  "name": "B",
  "scripts": {
    "build": "...",
    "test": "..."  
  },
  "metadata": "A"
}
```